src/lightning/fabric/strategies/launchers/xla.py

```diff
@@ -84,13 +84,12 @@ def _wrapping_function(
         return_queue: SimpleQueue,
         global_states: Optional[_GlobalStateSnapshot] = None,
     ) -> None:
-        self._strategy._local_rank = process_idx
```

Comment:
You say the process_idx here is actually the global rank. Looking at the spawn implementation, this index comes from `range(self._strategy.num_processes)`

But `num_processes` is the length per-node (host)

What am I missing?

If your hypothesis is correct, is it also a problem with regular DDP spawn? (non-XLA)

Reply:
The xla `spawn` replaces the index here: https://github.com/pytorch/xla/blob/cd7695196ea75387ea3e2f255b263cf7eafd7f4c/torch_xla/distributed/xla_multiprocessing.py#L323

So it is only issue for XLA